api/types.go

```diff
@@ -12,21 +15,31 @@ type ProxyConfig struct {
 }
 
 type OIDCConfig struct {
-	IssuerUrl    *string
-	ClientId     *string
-	ClientSecret *string
-	Audience     *string
+	IssuerUrl    *string `json:"issuer_url"`
```

Comment:
Is there some reason we use JSON for the config instead of YAML?
For user-orientated configs I find YAML might be more suited. cc @pb82 too for another view

Reply:
Don't have a preference here. For yaml we need to bring in another library (ghodss/yaml).

Reply:
I'd prefer using YAML if possible, as it is intended to be end-user configuration.

For YAML we have ghodss/yaml (https://github.com/ghodss/yaml) but also go-yaml(https://github.com/go-yaml/yaml). Is there some reason we need ghodss/yaml over go-yaml? The reason I am asking this is because go-yaml keeps being updated/maintained whereas ghodds/yaml seems to have been without any new tag since 2017

Reply:
I was always usign ghodss/yaml, but if go-yaml is better maintained, let's use that.